src/neo/Ledger/Blockchain.cs

```diff
@@ -496,15 +498,20 @@ private void Persist(Block block)
                         Transaction = tx
                     };
 
-                    snapshot.Transactions.Add(tx.Hash, state);
+                    clonedSnapshot.Transactions.Add(tx.Hash, state);
+                    clonedSnapshot.Transactions.Commit();
```

Comment:
When commit, all cached items will be traversed again. This reduces performance.

Reply:
Please have a look at this PR: #1619, which should be able to solve the problem.